### ISSUE TITLE
Relative date dashboard filter with default value is showing incorrect results as of 0.28

### ISSUE THREAD

**Issue by mazameli** [metabase/metabase#6994]
I'm opening this issue on behalf of @EvaSchreyer [based on our conversation here](https://github.com/metabase/metabase/issues/6059#issuecomment-364921601). Eva says this issue is keeping her from upgrading to 0.28.

Steps to reproduce:
- Create a SQL question with a field filter variable that is mapped to a date field
- Add that SQL question to a dashboard
- Add a relative date dashboard filter and connect it to the SQL question's field filter variable
- Give the relative date filter a default value of Past 30 days (this problem might not be limited to that exact default value, but it seems to be related to default values in general)
- Whenever you open the dashboard, the SQL card will not include yesterday's date

Here's how @evaschreyer describes the issue:

"When I open the dashboard in v0.25, the default filter value `last 30 days` works correctly and shows my data until yesterday. When I open the same dashboard in v0.28, I get only data until the day _before_ yesterday (even if data until yesterday can be selected when I use the filter "yesterday"). So I think it is the same issue again - as the filter works correctly, but the default value last 30 days is not working. "

"Also the new function "include today" does not change anything for the result in v0.28."

"Here are the relevant screenshots - same dashboard, same source:"

V0.25:
![bildschirmfoto 2018-02-13 um 10 28 52](https://user-images.githubusercontent.com/28557318/36142589-d59ac180-10a8-11e8-9c18-933c4911a7dc.png)

V0.28:
![bildschirmfoto 2018-02-13 um 10 29 09](https://user-images.githubusercontent.com/28557318/36142597-dca57e98-10a8-11e8-9843-d8fe72dc4d5b.png)

**Comment by paulozip** [comment]
Hi, guys. I'm facing a similar problem here.
After the update 0.28.1, when I try to set a date filter and select the current day, it’s not returning anything. I’ve run a SELECT NOW() just to make sure that the servers time is correct. Is anybody suffering from the same issue?

Here’s the question that I’m trying to use Date Filter
<img width="1425" alt="captura de tela 2018-03-02 as 10 42 20" src="https://user-images.githubusercontent.com/13841968/36902158-3f851bec-1e08-11e8-8468-aa6393dd2ac1.png">

**Comment by paulozip** [comment]
I think I found the problem. When Metabase runs a Date Filter parameter, with the `Current Day` selected, it uses the time as well in the parameter. For example, let's assume that I wanna select today's purchases. Selecting the `Today's date` on Metabase, it will run it as the following:

```
      SELECT *
          FROM purchases
      WHERE Date(`transactions`.`created`) = '2018-03-02 03:00:00'
```

Because of the '03:00:00', the query result won't return anything.

**Comment by paulozip** [comment]
up

**Comment by paulozip** [comment]
Guys, I found the problem. In a nutshell: timezone. I've set the timezone configuration on Metabase to "Database default" and it worked.

**Comment by oeon** [comment]
@paulozip great find! Making the change you suggest fixes some inconsistencies/incorrectness that I was also seeing. Thank you 🙂

**Comment by paulozip** [comment]
It's great to hear that, @oeon! I'm glad to help you. :D

**Comment by EvaSchreyer** [comment]
Hi guys, especially @paulozip, it works for me as well! I had to wait for a nightly upload to check the changes in the morning, but it looks good! thanks so much for digging into this, Paulo! 
@mazameli looks like (from a user perspective, not a technical perspective) this is a feasible workaround for this bug as my data are always one day old anyway (the way we update the database) and I therefore don´t rely on timezones. Not sure what setting the timezone to default does to live data though - I guess this is still something you guys have to dig into?
If I don´t encounter other issues I might update soon !! 🎉

**Comment by mazameli** [comment]
So am I right in saying that this isn't a bug, but rather a consequence of an incorrect timezone setting? If so, I'll go ahead and close this.

**Comment by EvaSchreyer** [comment]
Hi @mazameli, I am not sure. The timezone setting for Germany should be correct for my data - and it seems that @paulozip has a similar problem but probably with another timezone. If only the "not setting" of timezones gives me the correct default filter value (what was not the case in v0.25!), then I am not sure if its not still a bug and the not setting of the timezone a workaround?

**Comment by gizmodus** [comment]
@mazameli 
I have the same problems as @EvaScreyer with some additional problems with the date filter:

Timezone set to "Default":
- Set the date filter to "On" with today's date -> Correct results
- Set the date filter to "Previous 30 days" with "Include today" checked -> **Results of today are missing**

Timezone set to "Europe/Berlin":
- Set the date filter to "On" with today's date -> **Results of today are missing**
- Set the date filter to "Previous 30 days" with "Include today" checked -> **Results of today are missing**

**Comment by swatisinghi** [comment]
@mazameli I upgraded metabase from 0.25.2 to 0.29. I have a query running on redshift with date filters applied from April 2nd to April 5th. I am seeing a weird behaviour where once the query is running with correct date filters and returning data from April 2nd to April 5th and if I re-run it, it returns data from April 1st to April 4th.

**Comment by senior** [comment]
I've been doing some work on centralizing the code that handles timezone conversions in Metabase and I think I have reproduced this issue in a unit test. I don't have a fix yet, but will include one in https://github.com/metabase/metabase/pull/7520.

**Comment by vrosancic** [comment]
Hi all,

Has this fix been released? I am still having an issue with "Past 30 days including today" filter. It doesn't show records that have today's date.

Metabase version is 0.29.3.

Thanks!

**Comment by jornh** [comment]
@vrosancic it doesn’t look like. It’s merged to master *after* the date where 0.29.3 was released. 

So easiest option to try it out is probably the https://hub.docker.com/r/metabase/metabase-head/ Docker image (with its disclaimers). It’s currently built from master 4 days ago. Alternatively you will have to build from source or wait for the official v0.30 release.
